camera/Pilatus.py

```diff
@@ -214,12 +233,50 @@ def write_fill_mode(self, attr):
         gapfill = _getDictValue(self.__FillMode,data)
         _PilatusCamera.setGapfill(gapfill)
 
+#------------------------------------------------------------------
+#    Read cam_state attribute
+#------------------------------------------------------------------
+    def read_cam_state(self, attr):
+        status = _PilatusCamera.status()
+        status = _getDictKey(self.__CamStatus, status)
+        attr.set_value(status)
+
 #==================================================================
 #
 #    Pilatus command methods
 #
 #==================================================================
 
+#------------------------------------------------------------------
+#    sendCamserverCommand  command
+#------------------------------------------------------------------
+    @Core.DEB_MEMBER_FUNCT
+    def sendCamserverCmd(self, cmd):
+        _PilatusCamera.sendAnyCommand(cmd)
+
+#------------------------------------------------------------------
+#    set ReadoutGeometry command
+#------------------------------------------------------------------
+    @Core.DEB_MEMBER_FUNCT
+    def setReadoutGeometryCmd(self, rmode):
+        if _PilatusCamera.hasRoiCapability():
+            image = _CtControl.image()
+            roi = _getDictValue(self.__ReadoutRoi, rmode)
+            image.setRoi(roi)
+
+#------------------------------------------------------------------
+#    get ReadoutGeometry command
+#------------------------------------------------------------------
+    @Core.DEB_MEMBER_FUNCT
+    def getReadoutGeometryCmd(self):
+        if _PilatusCamera.hasRoiCapability():
+            image = _CtControl.image()
+            roi = image.getRoi()
```

Comment:
Hi @sebastien-petitdemange , I realize that method image.getHard().getRoi() does not exists?
What method do you suggest then to use to get the hardware Roi instead of the combination of Hard and Soft Rois?